### ISSUE TITLE
train / validation should report per-class IoU and dice

### ISSUE THREAD

**Issue by dbuscombe-usgs** [dbuscombe-usgs/segmentation_zoo#32]
- many problems are class-imbalanced
- in many images, many classes are often absent
- so we need better stats than mean iou and mean dice
- as a starting point , i suggest per-class iou and dice reporting
- perhaps also explore class weighting of iou scores

**Comment by dbuscombe-usgs** [comment]
a comparison of histograms seems appropriate for geographic imagery

thing that springs to my mind is the Kullback–Leibler distance, or mutual entropy, on the histograms of the two label images

this could also be a loss function ...

**Comment by dbuscombe-usgs** [comment]
` tf.keras.losses.KLDivergence `

loss = y_true * log(y_true / y_pred)

```
kl = tf.keras.losses.KLDivergence()
kl(y_true, y_pred).numpy()
```

**Comment by dbuscombe-usgs** [comment]
making this a new issue to explore

**Comment by dbuscombe-usgs** [comment]
Implement CM-generation and scores similar to https://github.com/zhiminwang1/Remote-Sensing-Image-Segmentation/blob/master/seg_metrics.py
